### PR TITLE
KAA-114

### PR DESCRIPTION
**Pull request by ashvayka**

### CODE REVIEW

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/view/appender/LogAppenderGrid.java [hunk 1]
```diff
@@ -65,7 +67,12 @@ public String getValue(LogAppenderDto item) {
                 new StringValueProvider<LogAppenderDto>() {
             @Override
             public String getValue(LogAppenderDto item) {
-                return item.getType().getLabel();
+                if (item.getType()==LogAppenderTypeDto.CUSTOM) {
```
Comment (izeigerman) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/view/appender/LogAppenderGrid.java]: Missing spaces near "=="

File: server/common/dao/src/main/avro/log_appender_parameters_schema.avsc [hunk 2]
```diff
@@ -94,6 +94,39 @@
...
+                    "name":"CustomAppenderParameters",
+                    "type":"record",
+                    "fields":[
+                        {
+                            "name":"name",
+                            "type":[
+                                {
+                                    "type":"string"
+                                },
+                                "null"
```
Comment (izeigerman) [inline server/common/dao/src/main/avro/log_appender_parameters_schema.avsc]: "type": [
"string",
"null"
]
would be enough

File: server/common/dto/src/main/java/org/kaaproject/kaa/common/dto/logs/avro/CustomAppenderParametersDto.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
```
Comment (izeigerman) [inline server/common/dto/src/main/java/org/kaaproject/kaa/common/dto/logs/avro/CustomAppenderParametersDto.java:69]: Missing braces

File: server/common/dto/src/main/java/org/kaaproject/kaa/common/dto/logs/LogAppenderInfoDto.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
```
Comment (izeigerman) [inline server/common/dto/src/main/java/org/kaaproject/kaa/common/dto/logs/LogAppenderInfoDto.java:101]: Missing braces

File: server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/logs/filesystem/FileSystemLogAppender.java [hunk 5]
```diff
@@ -59,17 +57,23 @@
     private boolean closed = false;
 
     public FileSystemLogAppender() {
+        logger = new LogbackFileSystemLogger();
     }
 
     public FileSystemLogAppender(String name) {
+        this();
```
Comment (izeigerman) [inline server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/logs/filesystem/FileSystemLogAppender.java:64]: Is it really necessary?

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/data/DataSource.java [hunk 6]
```diff
@@ -911,6 +914,23 @@ protected void onResult(Void result) {
...
+                protected void onResult(List<LogAppenderInfoDto> result) {
+                    appenderInfos.addAll(result);
+                }
+            });
+        } else {
+            if (callback != null) {
+                callback.onSuccess(appenderInfos);
+            }
+        }
+    }
```
Comment (izeigerman) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/data/DataSource.java:933]: Looks odd. Should we call "callback.onSuccess(appenderInfos)" after creation of "appenderInfos" object in the first case?

File: server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/logs/filesystem/FileSystemLogAppender.java (hunk 5 above)
Comment (ashvayka) [inline server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/logs/filesystem/FileSystemLogAppender.java:64]: Yes

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/data/DataSource.java (hunk 6 above)
Comment (ikulikov) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/data/DataSource.java:933]: No, callback.onSuccess will be called inside DataCallback when data will be retrieved.

**Comment by andreypanasenko** [comment]
LGTM

**Pull request merged**